packages/@n8n/nodes-langchain/nodes/agents/Agent/agents/ToolsAgent/execute.ts

```diff
@@ -93,7 +93,43 @@ export async function toolsAgentExecute(this: IExecuteFunctions): Promise<INodeE
 	const tools = (await getConnectedTools(this, true, false)) as Array<DynamicStructuredTool | Tool>;
 	const outputParser = (await getOptionalOutputParsers(this))?.[0];
 	let structuredOutputParserTool: DynamicStructuredTool | undefined;
+	/**
+	 * Handles the agent text output and transforms it in case of multi-output.
+	 *
+	 * @param steps - The agent finish or agent action steps.
+	 * @returns The modified agent finish steps or the original steps.
+	 */
+	function handleAgentFinishOutput(steps: AgentFinish | AgentAction[]) {
+		// Check if the steps contain multiple outputs
+		type AgentMultiOutputFinish = AgentFinish & {
+			returnValues: { output: Array<{ text: string; type: string; index: number }> };
+		};
+		const agentFinishSteps = steps as AgentMultiOutputFinish | AgentFinish;
```

Comment:
Should we make a copy here? Modifying function arguments may potentially lead to some hard-to-trace issues in the future.

Reply:
Cloning this object using `structuredClone` doesn't seem to work as it removes some class bindings Langchain requires so sticking with overriding the step content